packages/nimble-components/src/tooltip/specs/README.md

```diff
@@ -20,6 +20,17 @@ The nimble-tooltip project will first be implemented as a prototype, open issues
 
 [FAST tooltip API](https://github.com/microsoft/fast/blob/de7f234ef871204fcac2b5df59433d919809341d/packages/web-components/fast-foundation/src/tooltip/tooltip.spec.md)
 
+The tooltip will significantly differ from the FAST API in the way that a tooltip is associated with an "anchor" element. The FAST API provides two ways to associate a tooltip with the anchor: by `id` and by assigning the element directly to the tooltip's `anchorElement` property. We will internally make use of the latter, but the client will associate the two by wrapping them together in a container element:
+
+```
+<nimble-tooltip-connection>
+    <nimble-tooltip>...</nimble-tooltip>
+    <nimble-button>Hover To See Tooltip</nimble-button>
+</nimble-tooltip-connection>
```

Comment:
This seems super invasive to the dom structure.
My first instinct is instead this would be slotted into an element. Makes incrementally associating an element with a tooltip less intense:

```html
<nimble-button>Hover To See Tooltip</nimble-button>
```

```html
<nimble-button>Hover To See Tooltip
<nimble-tooltip slot="tooltip">Tooltip stuff</nimble-tooltip>
</nimble-button>
```

To brainstorm more alternatives, seems like a shortcut could be valuable as well. i.e. use the nimble-tooltip element and slot for advanced content or an attribute for text only content:

```html
<nimble-button nimble-tooltip="Just some text">Hover To See Tooltip</nimble-button>
```

Or maybe the user doesn't need to use the tooltip element directly and just slot some content:

```html
<nimble-button>Hover To See Tooltip
<span slot="tooltip">Tooltip stuff</span>
</nimble-button>
```

The nimble-tooltip could be interesting if there needs to be tooltip configuration that is user facing like popup direction or animation speed etc.

If you are choosing to brainstorm asyncronously in hld I'd like to see more discussion of pros and cons of alternatives.

Reply:
Another alternative: scope our immediate work to the use cases required by #857 (and possibly #858) while punting on the bigger work required to satisfy #309. This could mean either deferring the above API discussion altogether or it could mean having the discussion now but deferring the implementation.

To be clear, this is just an alternative, not a recommendation yet. But I want us to keep thinking about ROI. If we come up with a long-term tooltip solution that requires big architectural changes (e.g. forking all of FAST's templates) then we should pause and decide if now is the right time to do that.

Reply:
+1 for having more alternatives with their pros/cons discussed in the spec.

As part of that it would be good to have some high-level estimate of the amount of effort, maintenance burden, etc. related to the alternatives. For example, is there a common place to implement some of these patterns that involve slotting or new attributes?